internal/transport/http2_server.go

```diff
@@ -457,6 +459,22 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		return false
 	}
 
+	if parserError != nil {
+		if err := t.WriteStatus(
+			s,
+			status.Newf(codes.Internal, "transport: http2Server.operateHeaders failed parse headers: %v", parserError),
+		); err != nil && logger.V(logLevel) {
+			logger.Infof("transport: http2Server.operateHeaders failed to write status %v", err)
+		}
+		t.controlBuf.put(&cleanupStream{
```

Comment:
Is this okay after `WriteStatus` is called?  `WriteStatus` _may_ send a RST_STREAM itself.

Also my comment in the other PR may not be 100% correct - it's fine to not RST_STREAM as long as the incoming headers had the END_STREAM flag set on them.

Also the better code to use may be `ErrCodeNo` since I don't think the errors that get us here are HTTP/2 errors, and the status we write above will contain the error itself.

Reply:
I guess we can use earlyAbortStream. It simply sends the header and an end stream flag.